### PR TITLE
Fa to Pfa

### PR DESCRIPTION
**Pull request by saqzhao**
# Description

faToPfa creates a pFasta representation of a Fasta as one-hot encoding.

🐛 Bug Report

### Testing

None

### Checklist before requesting a review

- [ ] I performed a self-review of my code
- [ ] If it this a core feature, I added thorough tests
- [ ] The command `go fmt` or `make clean` was used on all files included

### CODE REVIEW

**Review by rimangan — APPROVED** [review]
Great work!

File: cmd/pFaTools/faToPfa.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		Start:  *start,
+		End:    *end,
+		Chrom:  *chrom,
+	}
+
+	pFaFaToPfa(s)
+}
+
+// pFaFaToPfa parses an input pFASTA file and converts the file according to user-defined settings.
+func pFaFaToPfa(s FaToPfaSettings) {
```
Comment (cfauci) [inline cmd/pFaTools/faToPfa.go:64]: what's going on with this function name? was this a typo?

File: fasta/pFasta/tools.go [hunk 2]
```diff
@@ -92,3 +91,67 @@ func Sample(input []PFasta, chrom string) fasta.Fasta {
...
+		if base == dna.A {
+			answer.Seq[idx] = pDna.Float32Base{A: 1, C: 0, G: 0, T: 0}
+		} else if base == dna.C {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 1, G: 0, T: 0}
+		} else if base == dna.G {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 1, T: 0}
+		} else if base == dna.T {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 0, T: 1}
+		} else if base == dna.N {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 0.25, T: 0.25}
```
Comment (cfauci) [inline fasta/pFasta/tools.go:118]: Was there a discussion about the reasoning for this being the probability for the N base?

File: fasta/pFasta/vcfToPfa_test.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package pFasta
+
```
Comment (cfauci) [inline fasta/pFasta/vcfToPfa_test.go:2]: is this a test that's still being worked on?

File: fasta/pFasta/tools.go (hunk 2 above)
Comment (craiglowe) [reply fasta/pFasta/tools.go:118]: Typo?

**Review by rimangan — APPROVED** [review]

File: fasta/pFasta/tools.go [hunk 4]
```diff
@@ -92,3 +48,67 @@ func Sample(input []PFasta, chrom string) fasta.Fasta {
...
+		if base == dna.A {
+			answer.Seq[idx] = pDna.Float32Base{A: 1, C: 0, G: 0, T: 0}
+		} else if base == dna.C {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 1, G: 0, T: 0}
+		} else if base == dna.G {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 1, T: 0}
+		} else if base == dna.T {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 0, T: 1}
+		} else if base == dna.N {
+			answer.Seq[idx] = pDna.Float32Base{A: 0, C: 0, G: 0, T: 0}
```
Comment (cfauci) [inline fasta/pFasta/tools.go:75]: Not saying this is a problem, but I'm curious why this is all zeros for N as opposed to 0.25 for everything? Was there an experimental reason or a theoretical reason?

**Review by cfauci — APPROVED** [review]

**Pull request merged**